app/routes.js

```diff
@@ -81,18 +92,27 @@ export const requireNoAuth = (api) => (nextState, replace) => {
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireNotVerified = (api) => (nextState, replace, cb) => {
-  api.user.get(function(err, user) {
-    if (err) {
-      // we expect a 401 Unauthorized when navigating to /email-verification
-      // when not logged in (e.g., in a new tab after initial sign-up)
-      if (err.status === 401) {
+export const requireNotVerified = (api, store) => (nextState, replace, cb) => {
+  let { blip: state } = store.getState();
+  if (state.loggedInUser) { 
+    checkIfVerified(state.loggedInUser);
+  } else {
+    api.user.get(function(err, user) {
+      if (err) {
+        // we expect a 401 Unauthorized when navigating to /email-verification
+        // when not logged in (e.g., in a new tab after initial sign-up)
+        if (err.status === 401) {
+          return cb();
+        }
+        throw new Error('Error getting user at /email-verification');
         return cb();
       }
-      throw new Error('Error getting user at /email-verification');
-      return cb();
-    }
-    if (user.emailVerified === true) {
+      checkIfVerified(user, cb);
+    });
+  }
+
+  function checkIfVerified(userToCheck) {
```

Comment:
Can you move this function to the top of `requireNotVerified` for better readability/consistency in fn declarations?